src/sentry/integrations/slack/message_builder/issues.py

```diff
@@ -72,6 +72,13 @@
     "bitbucket",
     "integrations:bitbucket",
 )
+SUPPORTED_CONTEXT_DATA = {
+    "Events": lambda group: get_group_global_count(group),
+    "Users Affected": lambda group: group.count_users_seen(),
+    "State": lambda group: SUBSTATUS_TO_STR.get(group.substatus, "").replace("_", " ").title(),
+    "First Seen": lambda group: time_since(group.first_seen),
+    "Regressed Date": lambda group: time_since(group.active_at),
+}
```

Comment:
Love this idea! For future extensibility and testing, we could go one step beyond and just assign these functions with a similar interface, so something like:

```

def events_handler(group: Group) -> str:
    # logic goes here

def users_affected_handler(group: Group) -> int:
  # logic goes here
 
...

# Define that the dict accepts a function handler that takes in a Group and returns any type
SUPPORTED_CONTEXT_DATA: Dict[str, Callable[Group], Any] = {
    "Events": events_handler,
    "Users Affected": users_affected_handler,
     ...
}
```

Now we can test and extend these functions without having to worry about modifying the dict in any way

Reply:
is this like replacing lambda functions with an actual function? these functions are pretty small which is why i put it in a lambda. it's just calling something on the `Group` object

Reply:
Yup exactly! And then we could add tests for those functions to make sure we have coverage. But agreed, there's small enough now and mostly we're just moving code so it's not a blocker